CanvasColorSpaceProposal.md

```diff
@@ -45,178 +45,172 @@ You can see this by making stripes of 1.0 and 0.0, and matching that pattern's b
+An ImageData's data attribute will be of the type indicated by the its storageFormat.
 
-### Limitations
-* toDataURL and toBlob may be lossy, depending on the file format, when used on a canvas that has an encoding other than "unorm8". Possible future improvements could solve or mitigate this issue by adding more file formats or adding options to specify the resource color space.
+Note that the storage formats are different from the pixel formats for 2D Canvas and WebGL pixel formats. This reflects the fact that there does not exist a Float16Array, although Float16 is a useful format for graphics buffers.
+
+### ImageBitmap
+
+ImageBitmaps are opaque bitmap objects that can be used by several consumers. ImageBitmaps should internally store their contents in the highest fidelity format possible.
+
+For example, if an ImageBitmap is created from a wide-color-gamut high-bit-depth image and is then drawn to a wide-color-gamut 2D canvas, the result should reflect the original image. In particular, the image should not have been clamped to sRGB or quantized to 8 bits anywhere along the way.
+
+TODO: createImageBitmap has a colorSpaceConversion flag that can be "none" or "default". The most useful behavior is "respect the color space and convert appropriately when drawing to a canvas", which is the behavior of "default" on some browsers. A less useful but sometimes desirable behavior is "ignore the color space, and reinterpret the pixels as though they were in sRGB", which is the behavior of "none". This behavior should be more clearly specified.
 
 ### Adoption
+
 Lack of color management and color interoperability is a longstanding complaint about the canvas API.
 Authors of games and imaging apps are expected to be enthusiastic adopters.
 
-## Unresolved Issues
+## Issues
 
-* Should we support custom color spaces based on ICC profiles? Would offer ultimate flexibility. Would be hard to make implementations as efficient as built-in color spaces, in particular for implement linearPixelMath for profiles that have arbitrary transfer curves.
+Issues:
 
-* Should float16 allow alpha values outside [0,1] range at any stage of the pipeline? What would they mean?
+* How should feature detection work? The combinations of CanvasColorSpace and CanvasColorType supported by 2D Canvas and WebGL are, for now, obviously "all of them", but this is not likely to remain the case.
 
-* Should there be API-level support for mixing chromaticities and transfer functions, including use of no-op transfer functions?
+* Should we unify ImageDataStorageFormat with CanvasColorType? This might create more hassle than it's worth, because there are types for CanvasColorType (e.g, float16) that cannot exist for ImageDataStorageFormat at this time.
+
+* For ImageData, is it appropriate to be able to query the ImageDataStorageFormat, when that can be inferred from the type of the data member? This would make sense if we supported both uint8 and unorm8.
```

Comment:
What is `uint8` that isn't actually `unorm8`?

Reply:
Integer-based texture formats (GL_RGBA8I, MTLPixelFormatRGBA8Uint, etc).

From the perspective of someone using 2D canvas, "unorm" is a foreign concept. In WebGL, it makes sense.

But I'll do this separately.